### PR TITLE
refactor: accounts state access

### PR DESCRIPTION
**Pull request by dessaya**

### CODE REVIEW

File: packages/vm/core/accounts/state.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+			}
+			return isc.NewContractAgentID(chainID, hn), nil
+		case common.AddressLength:
+			var ethAddr common.Address
+			copy(ethAddr[:], []byte(key))
+			return isc.NewEthereumAddressAgentID(chainID, ethAddr), nil
+		default:
+			panic("bad key length")
+		}
+	}
```
Comment (jorgemmsilva) [inline packages/vm/core/accounts/state.go:122]: isn't this missing L1addresses?

**Review by jorgemmsilva — APPROVED** [review]

File: packages/vm/core/accounts/state.go (hunk 1 above)
Comment (dessaya) [reply packages/vm/core/accounts/state.go:122]: Idk, I didn't write this, I just moved it.

Seems like it is missing? But in that case the bug is also on develop :thinking:

Comment (dessaya) [reply packages/vm/core/accounts/state.go:122]: Unlike ContractAgentID and EthereumAgentID, AddressAgentID does not include a ChainID, that's why.

Comment (jorgemmsilva) [reply packages/vm/core/accounts/state.go:122]: that gets handled by `codec.AgentID.Decode`, nevermind :+1:

**Pull request merged**